backend/entities/courses/section_entity.py

```diff
@@ -32,7 +34,7 @@ class SectionEntity(EntityBase):
 
     # Term the section is in
     # NOTE: This defines a one-to-many relationship between the term and sections tables.
-    term_id: Mapped[int] = mapped_column(ForeignKey("courses__term.id"))
+    term_id: Mapped[str] = mapped_column(ForeignKey("courses__term.id"))
     term: Mapped["TermEntity"] = relationship(back_populates="course_sections")
 
```

Comment:
@KrisJordan interesting!! I will take a look and play around with this. I do wonder about importing the user entity from courses, as this was an issue previously.

Reply:
@KrisJordan after a while of trying things out, this works!! :

```py
staff: Mapped["UserSectionEntity"] = relationship(
        "UserEntity",
        secondary="join(UserSectionEntity, UserEntity, UserSectionEntity.user_id == UserEntity.id)",
        primaryjoin="SectionEntity.id == UserSectionEntity.section_id",
        secondaryjoin="UserSectionEntity.user_id == UserEntity.id",
        viewonly=True,
    )
```

Reply:
Will test and add more functionality, but now querying the entity here returns a list of user entities based on the secondary table! So, SQLAlchemy said we cannot pass in an entity directly in `secondary` (only `Table`s), so I had to use `join()` to manually stitch one together.

Reply:
As you have probably seen, this actually still does not fully work because the member_type is not passed over. I wrote a pure SQL query that accomplishes this, however I have tried numerous ways to do this in SQLAlchemy and have not had any success. I will try to implement a database relationship like normal here and see if I can get that working. It may not be the most ideal, but the SQLAlchemy relationship() seems to only want to take in entities, not custom models.